### PR TITLE
inpainting, specify an initial frame and the video will be extrapolat…

### PR DESCRIPTION
**Pull request by nagolinc**
…ed from there

inpainting_Screenshot 2023-03-27 091243

https://user-images.githubusercontent.com/7775917/228009298-6bcbae51-f58f-4da9-b781-80270b54e026.mp4

### CODE REVIEW

File: scripts/modelscope-text2vid.py [hunk 1]
```diff
@@ -140,6 +143,61 @@ def process(skip_video_creation, ffmpeg_location, ffmpeg_crf, ffmpeg_preset, fps
...
+            #padding_width = [(0, 0), (0, 0), (0, frames-inpainting_frames), (0, 0), (0, 0)]
+            #padded_latents = np.pad(image_latents, pad_width=padding_width, mode='constant', constant_values=0)
+
+            latent_h=height//8
+            latent_w=width//8
+            latent_noise=np.random.normal(size=(1,4,frames,latent_h,latent_w))
+            mask=np.ones(shape=(1,4,frames,latent_h,latent_w))
+
+            if len(inpainting_weights)>0:
+                mask_weights=eval(inpainting_weights)
```
Comment (kabachuha) [inline scripts/modelscope-text2vid.py:185]: Please, don't use `eval`. It's absolutely unsafe. If you have to specify the weights dynamically, take a looks how it's done in Deforum with a more safe library `numexpr` https://github.com/deforum-art/deforum-for-automatic1111-webui/blob/automatic1111-webui/scripts/deforum_helpers/animation_key_frames.py

**Review by kabachuha — CHANGES_REQUESTED** [review]

File: scripts/modelscope-text2vid.py (hunk 1 above)
Comment (nagolinc) [reply scripts/modelscope-text2vid.py:185]: can I just use safeeval (https://github.com/viur-framework/safeeval) or just put a warning that says "don't paste untrusted code into this box"?

**Review by kabachuha — APPROVED** [review]
I think I'll merge it with the basic functionality of setting up the first frame and then we'll come up with a better solution for weights setup and selecting multiple images (JSON?)

**Comment by Natotela** [comment]
so excuse me for the dumb question, but this is from April, says merged, and I got today's most updated extension, so how come I don't have that tab?

**Comment by kabachuha** [comment]
It's inside the txt2img tab, at the bottom

**Comment by Natotela** [comment]
LoL, involuntarily made a shutterstock ad (watermark)

https://github.com/kabachuha/sd-webui-text2video/assets/33201750/fb098e70-694d-4531-9396-84678fe57948

**Pull request merged**